waspc/data/Generator/templates/react-app/src/vitest.helpers.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import { BrowserRouter as Router } from 'react-router-dom'
+import { render } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { beforeAll, afterEach, afterAll } from 'vitest'
+import { Query } from './queries'
+import config from './config'
+
+export function renderWrapped(ui: React.ReactElement): any {
+  const client = new QueryClient()
+  const { rerender, ...result } = render(
+    <QueryClientProvider client={client}><Router>{ui}</Router></QueryClientProvider>
+  )
+  return {
+    ...result,
+    rerender: (rerenderUi: React.ReactElement) =>
+      rerender(
+        <QueryClientProvider client={client}><Router>{rerenderUi}</Router></QueryClientProvider>
+      )
+  }
+}
+
+export const server = setupServer()
+
+export function initCallbacks() {
+  beforeAll(() => server.listen())
+  afterEach(() => server.resetHandlers())
+  afterAll(() => server.close())
+}
+
+export function mockQuery(query: Query<any, any>, resJson: any): void {
+  const route = query.queryCacheKey[0]
```

Comment:
This bit feel a bit fragile - using the cache key abstraction to get the path of the query.

I would suggest that we add another key on the query called `route` or similar (it can have the same value) but then the expectation is clear and changing query keys won't affect testing.

Reply:
True, we could explicitly track the route separately. Will do